### ISSUE TITLE
TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by akkariyassine** [kucherenko/jscpd#421]
**Describe the bug**
jscpd crashes in the middle of analysis, stacktrace below

TypeError: Cannot read property 'length' of undefined
    at calculateLocation (C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\tokenizer\dist\tokenize.js:72:59)
    at C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\tokenizer\dist\tokenize.js:115:32
    at Array.map (<anonymous>)
    at tokenize (C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\tokenizer\dist\tokenize.js:115:10)
    at Object.createTokenMapBasedOnCode (C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\tokenizer\dist\tokenize.js:120:20)
    at Tokenizer.generateMaps (C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\tokenizer\dist\index.js:12:27)
    at Detector.<anonymous> (C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\core\dist\detector.js:33:46)
    at Generator.next (<anonymous>)
    at C:\Users\AppData\Roaming\npm\node_modules\jscpd\node_modules\@jscpd\core\dist\detector.js:8:71
    at new Promise (<anonymous>)

**Comment by kucherenko** [comment]
Thank you for the report!
Could you please provide more information about source for analysis? What is the language of data (js, php, etc.), how big it is, etc.

**Comment by hata6502** [comment]
https://github.com/hata6502/editorjs-layout/runs/1824435451?check_suite_focus=true

**Comment by hata6502** [comment]
I'm experiencing the same error: 

```
$ git clone https://github.com/codex-team/editor.js
$ cd editor.js
$ npx jscpd
Clone found (javascript):
 - src/components/tools/paragraph/node_modules/@babel/core/node_modules/ms/index.js [1:1 - 162:2] (161 lines, 1061 tokens)
   src/components/tools/paragraph/node_modules/@babel/traverse/node_modules/ms/index.js [1:1 - 162:2]

TypeError: Cannot read property 'length' of undefined
    at calculateLocation (/home/hata/jscpd/packages/tokenizer/src/tokenize.ts:85:55)
    at /home/hata/jscpd/packages/tokenizer/src/tokenize.ts:140:25
    at Array.map (<anonymous>)
    at tokenize (/home/hata/jscpd/packages/tokenizer/src/tokenize.ts:139:6)
    at Object.createTokenMapBasedOnCode (/home/hata/jscpd/packages/tokenizer/src/tokenize.ts:148:28)
    at Tokenizer.generateMaps (/home/hata/jscpd/packages/tokenizer/src/index.ts:11:12)
    at Detector.<anonymous> (/home/hata/jscpd/packages/core/src/detector.ts:29:52)
    at Generator.next (<anonymous>)
    at /home/hata/jscpd/packages/core/src/detector.ts:8:71
    at new Promise (<anonymous>)
```

https://github.com/kucherenko/jscpd/blob/master/packages/tokenizer/src/tokenize.ts#L85
It seems this error occurs when `lines` is an empty array. 

It occurs in `node_modules`, so it can be resolved by using `-g, --gitignore` option.

**Comment by hata6502** [comment]
https://github.com/cucumber/cucumber-jvm

```
$ gh repo clone cucumber/cucumber-jvm
Cloning into 'cucumber-jvm'...
remote: Enumerating objects: 438, done.
remote: Counting objects: 100% (438/438), done.
remote: Compressing objects: 100% (270/270), done.
remote: Total 101460 (delta 146), reused 347 (delta 100), pack-reused 101022
Receiving objects: 100% (101460/101460), 25.48 MiB | 1.10 MiB/s, done.
Resolving deltas: 100% (45770/45770), done.
$ cd cucumber-jvm/
$ jscpd
Clone found (java):
 - examples/java-wicket/java-wicket-main/src/main/java/io/cucumber/examples/java/wicket/view/Create.java [1:1 - 6:32] (5 lines, 71 tokens)
   examples/java-wicket/java-wicket-main/src/main/java/io/cucumber/examples/java/wicket/view/Rent.java [1:1 - 6:31]

TypeError: Cannot read property 'length' of undefined
    at calculateLocation (/usr/local/lib/node_modules/jscpd/node_modules/@jscpd/tokenizer/dist/tokenize.js:72:59)
    at /usr/local/lib/node_modules/jscpd/node_modules/@jscpd/tokenizer/dist/tokenize.js:115:32
    at Array.map (<anonymous>)
    at tokenize (/usr/local/lib/node_modules/jscpd/node_modules/@jscpd/tokenizer/dist/tokenize.js:115:10)
    at Object.createTokenMapBasedOnCode (/usr/local/lib/node_modules/jscpd/node_modules/@jscpd/tokenizer/dist/tokenize.js:120:20)
    at Tokenizer.generateMaps (/usr/local/lib/node_modules/jscpd/node_modules/@jscpd/tokenizer/dist/index.js:12:27)
    at Detector.<anonymous> (/usr/local/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:33:46)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:8:71
    at new Promise (<anonymous>)
```

**Comment by kucherenko** [comment]
fixed at jscpd@3.3.24
